config/routes.rb

```diff
@@ -210,6 +210,12 @@
       post :request_membership
       get :isa_children
       get :overview
+      post :update_investigation_permission
+      post :update_study_permission
+      get :investigation_shared_with
+      get :study_shared_with
+      post :upload_project_file
+      get :get_file_list
```

Comment:
Some of these are related to new methods.
For example, 'upload_project_file' is for uploading files to the new folder structure (Slideshows, Articles, Posters, and Presentations).

I can use JSON API instead of using 'investigation_shared_with' and 'study_shared_with' but in that case, multiple GET req will be produced to get each person's name to whom the investigation is shared.

Do you suggest to add all these methods to the JSON API? 
Or just add methods associated with the new features to JSON API + use multiple GET req to retrieve people's names?